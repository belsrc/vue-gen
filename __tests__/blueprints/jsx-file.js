const genJsx = require('./../../src/blueprints/jsx-file');

const compName = 'TestComponent';
const className = 'test-component';

describe('genJsx', () => {
  test('has the expected ouput structure not as page and without state', () => {
    const actual = genJsx(compName, className, false, false);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure as page and with state', () => {
    const actual = genJsx(compName, className, true, true);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure as page without state', () => {
    const actual = genJsx(compName, className, true, false);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure not as page with state', () => {
    const actual = genJsx(compName, className, false, true);

    expect(actual).toMatchSnapshot();
  });
});
