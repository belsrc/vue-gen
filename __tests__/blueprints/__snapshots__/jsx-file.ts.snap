// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genJsx has the expected ouput structure as page and with state 1`] = `
"import './styles.scss';

import { mapState, mapGetters, mapActions } from 'vuex';

export default {
  name: 'TestComponent',

  props: {},

  data() {
    return {};
  },

  computed: {
    ...mapState({}),

    ...mapGetters({}),
  },

  watch: {},

  async created() {
    return await this.loadPageData();
  },

  mounted() {},

  // When route changes and this component is already rendered, this guard is used
  async beforeRouteUpdate(to, from, next) {
    await this.loadPageData();

    return next();
  },

  methods: {
    ...mapActions({}),

    async loadPageData() {},
  },

  render() {
    return (
      <div class='test-component'>
        (ಠ益ಠ)
      </div>
    );
  },
};
"
`;

exports[`genJsx has the expected ouput structure as page without state 1`] = `
"import './styles.scss';

export default {
  name: 'TestComponent',

  props: {},

  data() {
    return {};
  },

  computed: {},

  watch: {},

  async created() {
    return await this.loadPageData();
  },

  mounted() {},

  // When route changes and this component is already rendered, this guard is used
  async beforeRouteUpdate(to, from, next) {
    await this.loadPageData();

    return next();
  },

  methods: {
    async loadPageData() {},
  },

  render() {
    return (
      <div class='test-component'>
        (ಠ益ಠ)
      </div>
    );
  },
};
"
`;

exports[`genJsx has the expected ouput structure not as page and without state 1`] = `
"import './styles.scss';

export default {
  name: 'TestComponent',

  props: {},

  data() {
    return {};
  },

  computed: {},

  watch: {},

  created() {},

  mounted() {},

  methods: {},

  render() {
    return (
      <div class='test-component'>
        (ಠ益ಠ)
      </div>
    );
  },
};
"
`;

exports[`genJsx has the expected ouput structure not as page with state 1`] = `
"import './styles.scss';

import { mapState, mapGetters, mapActions } from 'vuex';

export default {
  name: 'TestComponent',

  props: {},

  data() {
    return {};
  },

  computed: {
    ...mapState({}),

    ...mapGetters({}),
  },

  watch: {},

  created() {},

  mounted() {},

  methods: {
    ...mapActions({}),
  },

  render() {
    return (
      <div class='test-component'>
        (ಠ益ಠ)
      </div>
    );
  },
};
"
`;
